Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -201,122 +203,137 @@ struct SymbolGraphLoader {
+                                    continue
+                                }
+                                // The symbols does not contains any information for the fallback platform
+                                inheritedAvailability.domain = SymbolGraph.Symbol.Availability.Domain(rawValue: fallbackPlatform.rawValue)
+                                symbolAvailability.append(inheritedAvailability)
+                                if inheritedAvailability.introducedVersion != nil {
+                                    platformsAvailable.insert(fallbackPlatform.rawValue)
                                 }
                             }
                         }
                     }
-                    // Add the missing default platform availability.
-                    missingAvailabilities.forEach { missingAvailability in
-                        if !symbolAvailability.contains(missingAvailability.platformName) {
-                            guard let defaultAvailability = AvailabilityItem(missingAvailability) else { return }
-                            symbolAvailability.availability.append(defaultAvailability)
-                        }
-                    }
-                    symbol.mixins[selector]![SymbolGraph.Symbol.Availability.mixinKey] = symbolAvailability
                 }
-            }
-        }
-    }    
-
-    /// If the bundle defines default availability for the symbols in the given symbol graph
-    /// this method adds them to each of the symbols in the graph.
-    private func addDefaultAvailability(to symbolGraph: inout SymbolGraph, moduleName: String) {
-        // Check if there are defined default availabilities for the current module
-        if let defaultAvailabilities = bundle.info.defaultAvailability?.modules[moduleName],
-            let platformName = symbolGraph.module.platform.name.map(PlatformName.init) {
-
-            // Prepare a default availability versions lookup for this module.
-            let defaultAvailabilityVersionByPlatform = defaultAvailabilities
-                .reduce(into: [PlatformName: SymbolGraph.SemanticVersion](), { result, defaultAvailability in
-                    if let introducedVersion = defaultAvailability.introducedVersion, let version = SymbolGraph.SemanticVersion(string: introducedVersion) {
-                        result[defaultAvailability.platformName] = version
+                
+                // Add the module default availability information.
+                defaultAvailabilities.forEach { defaultAvailability in
+                    
+                    // Check that if there was a symbolgraph for this platform, the symbol was present on it,
+                    // if not it means that the symbol is not available for the default platform.
+                    guard symbolsByPlatformName[defaultAvailability.platformName.rawValue]?.contains(where: {
+                        symbolID == $0.precise
+                    }) != false else {
```

Comment:
minor: AFAICT this is the only place that uses `symbolsByPlatformName`. If that's correct, maybe it would be worth using a set of symbol IDs so that this can use `contains(_:)` instead of `contains(where:)`.

In other words, changing `symbolsByPlatformName` from `[String: [SymbolGraph.Symbol.Identifier]]` to [String: Set<String>]